publication-commons/src/main/java/no/unit/nva/publication/service/impl/UpdateResourceService.java

```diff
@@ -144,18 +149,61 @@ public ImportCandidate updateImportCandidate(ImportCandidate importCandidate) th
         throw new BadRequestException("Can not update already imported candidate");
     }
 
-    public void unpublishPublication(Publication publication) {
+    public void unpublishPublication(Publication publication,
+                                     Stream<TicketEntry> existingTicketStream,
+                                     UnpublishRequest unpublishRequest) {
         publication.setStatus(UNPUBLISHED);
         publication.setDuplicateOf(publication.getDuplicateOf());
         publication.setModifiedDate(clockForTimestamps.instant());
         var resource = Resource.fromPublication(publication);
 
+
+        var existingTicketTransactionsItems =
+            existingTicketStream
+                .filter(this::isPendingTicket)
+                .map(this::setToNotApplicable)
+                .map(Entity::toDao)
+                .map(dao -> (TicketDao) dao)
+                .map(this::createPutTransactionItems)
+                .toList();
+        var unpublishinRequestTransationItems =
+            new UnpublishRequestDao(unpublishRequest).createInsertionTransactionRequest().getTransactItems();
         var updateResourceTransactionItem = updateResource(resource);
 
-        var request = new TransactWriteItemsRequest().withTransactItems(updateResourceTransactionItem);
+        var transactionItems = new ArrayList<>(existingTicketTransactionsItems);
+        transactionItems.add(updateResourceTransactionItem);
+        transactionItems.addAll(unpublishinRequestTransationItems);
+
+        var request = new TransactWriteItemsRequest().withTransactItems(transactionItems);
         sendTransactionWriteRequest(request);
     }
 
+    private boolean isPendingTicket(TicketEntry ticketEntry) {
+        return TicketStatus.PENDING.equals(ticketEntry.getStatus());
+    }
+
+    private TransactWriteItem createPutTransactionItems(TicketDao ticketDao) {
```

Comment:
@brinxmat kan du sjekke en ekstra gang på at akkurat denne biten ble riktig?
Testene kjøres igjennom, men dette er litt magisk for meg.

Reply:
Sjekket ut egen branch i sandbox, det ser bra ut.